File: app/src/main/java/me/devsaki/hentoid/util/ContentNotRemovedException.java
package me.devsaki.hentoid.util;

import me.devsaki.hentoid.database.domains.Content;

public class ContentNotRemovedException extends Exception {

    private final Content content;

    public ContentNotRemovedException(Content content, String message)
    {
        super(message);
        this.content = content;
    }

    public Content getContent() {
        return content;
    }
}

File: app/src/main/java/me/devsaki/hentoid/util/exception/ContentNotRemovedException.kt
package me.devsaki.hentoid.util.exception

import me.devsaki.hentoid.database.domains.Content

class ContentNotRemovedException : Exception {

    val content: Content

    constructor(content: Content, message: String, cause: Throwable) : super(message, cause) {
        this.content = content
    }

    constructor(content: Content, message: String) : super(message) {
        this.content = content
    }
}
